### PR TITLE
Supermatter Beacon replacing Shard in Cargo

### PR DESCRIPTION
**Pull request by Cheshify**
## About The Pull Request

This PR adds a Supermatter Beacon, an activatable item that call down an anchored Supermatter Shard to it's position after a 5 second delay. Your new Supermatter Shard is guaranteed to come to your station in a safe bluespace pod after a comfortable delay to vacate the position of the beacon, and it will already be anchored to the floor below, though it can still be moved with a wrench.

image
image

## Why It's Good For The Game

The purpose of this PR is primarily safety. A properly handled Supermatter Beacon is a lot safer to handle than a Shard, and no longer will you be forced to drag and then dust yourself trying to get a shard where you want it, whether that be for engine building, or otherwise.

## Changelog
:cl:
add: Added a Supermatter Beacon, purchasable from cargo, it can call a shard down to it's position!
tweak: Supermatter Shard crates now contain the Beacon, rather than the shard itself.
/:cl:

### CODE REVIEW

**Comment by MrDoomBringer** [comment]
👎 part of the fun is dragging a dangerous shard around
opens up emergent gameplay re: traitors intercepting the shard shipment etc etc

**Comment by Swiftfeather** [comment]
> 👎 part of the fun is dragging a dangerous shard around
> opens up emergent gameplay re: traitors intercepting the shard shipment etc etc

it ALSO opens up the possibility of buying like 10 of these and spamming random supermatters around the station

**Comment by Cheshify** [comment]
> 👎 part of the fun is dragging a dangerous shard around
> opens up emergent gameplay re: traitors intercepting the shard shipment etc etc

This can literally still happen, just whack the person transporting the shard, grab the beacon (which they NEED to have in a crate or in their hand), and run off with it.

**Comment by itseasytosee** [comment]
How are you supposed to know they have the beacon when you can just stuff it in a backpack? Where's the danger in putting an item into a backpack.

**Comment by Cheshify** [comment]
> How are you supposed to know they have the beacon when you can just stuff it in a backpack? Where's the danger in putting an item into a backpack.

Read the PR, item is bulky, and can't be put into a backpack, as far as I'm aware.

**Comment by itseasytosee** [comment]
What makes this beacon bulkier than other beacons?

**Comment by Cheshify** [comment]
> What makes this beacon bulkier than other beacons?

w_class = WEIGHT_CLASS_BULKY??? If you mean what makes it physically bulkier than other beacons, I could probably find some new icon for it, if that's what you want.

**Comment by itseasytosee** [comment]
Just seems like !fun! removal to me.

**Comment by D4C-420** [comment]
it starts anchored which is the biggest issue here. you cant move it without risking the act of clicking it with a wrench

File: code/modules/cargo/sm_beacon.dm [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+	icon_state = "beacon"
+	lefthand_file = 'icons/mob/inhands/misc/devices_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/misc/devices_righthand.dmi'
+	w_class = WEIGHT_CLASS_BULKY
+	var/droptype = /obj/machinery/power/supermatter_crystal/shard/anchored
+
+/obj/item/dropbeacon/sm_beacon/proc/calldown_pod()
+	var/obj/structure/closet/supplypod/bluespacepod/pod = new()
+	new droptype(pod)
+	pod.explosionSize = list(0,0,0,0)
```
Comment (Fikou) [inline code/modules/cargo/sm_beacon.dm:15]: BOOOOOOO

**Comment by Tlaltecuhtli** [comment]
if  you want to use a beacon use the express console for it instead of making this

**Comment by 81Denton** [comment]
> 👎 part of the fun is dragging a dangerous shard around
> opens up emergent gameplay re: traitors intercepting the shard shipment etc etc

What Doombringer said, the shard being incredibly dangerous to move around is just part of the fun. Whether it's people blindly running into it, atmos pushing it around, someone with a grav catapult mech throwing it etc.

**Comment by MrDoomBringer** [comment]
> if  you want to use a beacon use the express console for it instead of making this

this is the correct response
i amaze even myself sometimes

**Comment by Cheshify** [comment]
You can... still drag a shard around? If you're dragging a shard, you're going to be standing next to it, meaning you can wrench it? This PR doesn't change too much for griefing or antagging with an SM shard, it's purpose is to make the act of setting up a replacement, or 2nd engine, faster and safer. If you desperately want a shard to grief with, you can still buy it and just unwrench it.

**Comment by itseasytosee** [comment]
Faster and safer just mean less engaging and less fun.

**Comment by Cheshify** [comment]
> Faster and safer just mean less engaging and less fun.

Less fun for... griefing, which is one small part of a large game

**Comment by MrDoomBringer** [comment]
you say "safer" as if that word is a good thing. As if it doesn't say "metal death trap masquerading as a space station" on the game's website

Also, again, this functionality literally already exists via the express supply console. There is no need for this PR

**Comment by Cheshify** [comment]
> you say "safer" as if that word is a good thing. As if it doesn't say "metal death trap masquerading as a space station" on the game's website
> 
> Also, again, this functionality literally already exists via the express supply console. There is no need for this PR

The functionality that involves setting up an explosive pod within a preexisting supermatter setup, this is only genuinely safer for people wanting to make engines, for any other purpose, it's the same as any other shard.

**Comment by itseasytosee** [comment]
That's his point cheshify, setting up the engine shouldn't be safe.

**Comment by Cheshify** [comment]
> That's his point cheshify, setting up the engine shouldn't be safe.

Dusting yourself on an SM because you need to set something up is not fun.

**Comment by itseasytosee** [comment]
That's where you and I disagree, friend.

**Comment by D4C-420** [comment]
> This PR adds a Supermatter Beacon, an activatable item that call down an anchored Supermatter Shard to it's position after a 5 second delay.

The thing is anchored in a single spot. You can't drag it around because its anchored in place. Unless in this one specific case anchored means something different than what it normally does you **CAN'T** move the shard at all without taking a wrench and clicking on it which most people avoid doing because they don't want to mess up and get dusted which 1.  is something your against and 2 because it's harder to move at all once placed which ruins the fun of an sm shard. People moving the sm shard once its opened it a big part of the risk in setting up a new sm.

**Comment by Cheshify** [comment]
> > This PR adds a Supermatter Beacon, an activatable item that call down an anchored Supermatter Shard to it's position after a 5 second delay.
> 
> The thing is anchored in a single spot. You can't drag it around because its anchored in place. Unless in this one specific case anchored means something different than what it normally does you **CAN'T** move the shard at all without taking a wrench and clicking on it which most people avoid doing because they don't want to mess up and get dusted which 1. is something your against and 2 because it's harder to move at all once placed which ruins the fun of an sm shard. People moving the sm shard once its opened it a big part of the risk in setting up a new sm.

Please tell me how you're going to move a shard without dragging it, which means being right next to the sm, which means being close enough to wrench it. With this, nobody NEEDS to touch a proper SM setup but the people who are ACTIVELY SABOTAGING shit.

**Comment by D4C-420** [comment]
> Please tell me how you're going to move a shard without dragging it, which means being right next to the sm, which means being close enough to wrench it. With this, nobody NEEDS to touch a proper SM setup but the people who are ACTIVELY SABOTAGING shit.

Im pointing out that you CAN'T drag the shard which is the problem. You or anyone else should be able to move it.  There is rarely anything dangerous about a sm set up directly. You don't even need to touch the sm when sabotaging it so a shard being stuck in place makes no sense. A sm shard is meant to be dangerous. You take away from what people can do by making it so risk free to handle by making it harder to steal, making it harder to move, and making only the CE and captain able to open it (remember outside of mind shield implants cargo can open any crate by force). Also if its unanchored you can move it in the same way anything other unanchored objects can be move meaning gravity catapults,   a few chems and bombs at the right distance.

**Comment by itseasytosee** [comment]
@Cheshify Is this an ided scenario? because it's beginning to look like it.

**Comment by Cheshify** [comment]
I'm done arguing about this PR, please actually look at what it does, and consider what it can do for actual engineers, not for griefing.

**Comment by MrDoomBringer** [comment]
@Cheshify you know you're supposed to wear magboots if you want to safely drag around a shard, right?

**Comment by D4C-420** [comment]
> I'm done arguing about this PR, please actually look at what it does, and consider what it can do for actual engineers, not for griefing.

Still trying to figure that out cause now only CE and captain can open up an sm shard so if there is no CE this just takes away the ability for regular old engis and atmoses to make new engines.

**Comment by Cheshify** [comment]
> Still trying to figure that out cause now only CE and captain can open up an sm shard so if there is no CE this just takes away the ability for regular old engis and atmoses to make new engines.

Uh... I didn't change that, please actually read what I changed?

**Comment by Mothblocks** [comment]
i think its funny

**Pull request closed without merging**